src/utilities.jl

```diff
@@ -0,0 +1,77 @@
+function checkbounds(A::AbstractRFP, i::Integer, j::Integer)
+    n, k, l = _rfpsize(A)
+    0 < i ≤ n && 0 < j ≤ n || throw(BoundsError("[$i, $j] isn't in an $n × $n matrix"))
+    return n, k, l
+end
+
+"""
+    _packedinds(A::AbstractRFP, i, j, neven, l)
+
+Returns an index tuple into `A.data` where `ij` is in the correct index range and correct triangle
+
+The triangle is checked before entry to this function because `ij` not being in the `A.uplo` triangle
+is handled differently for triangular, which returns `zero(T)` and Hermitian, which returns the
+conjugate of `A[j, i]`.
+
+`l`, the smaller dimension of the parent array, and `neven`, whether the virtual
+size `n` is even, are already calculated and are passed to this function
+"""
+function _packedinds(A::AbstractRFP, i::Integer, j::Integer, neven::Bool, l::Int)
+    return _packedinds(Int(i), Int(j), A.uplo == 'L', neven, A.transr == 'N', l)
+end
+
+function _packedinds(i::Int, j::Int, lower::Bool, neven::Bool, tr::Bool, l::Int)
+    if lower
+        conj = l < j
+        inds = conj ?  (j - l, i + !neven - l) : (i + neven, j)
+    else
+        conj = (j + !neven) ≤ l
+        inds = conj ? (l + neven + j, i) : (i, j + !neven - l)
+    end
+    return tr ? (inds, conj) : (reverse(inds), !conj)
+end
+
+"""
+    _parentsize(n::Integer)
+
+Returns the size of the `data` field in an RFP array with `transr == 'N'`, representing a matrix of size `n`
+"""
+function _parentsize(n::Integer, tr::Bool = false)
+    n > 0 || throw(ArgumentError("n = $n must be positive"))
+    sz = (n + iseven(n), (n + 1) >> 1)
+    return tr ? reverse(sz) : sz
+end
+
+Base.parent(A::AbstractRFP) = A.data
```

Comment:
Let's also drop this one

Reply:
There were a couple of 'if's missed in the editing of `src/triangular.jl` and it took me an embarrassingly long time to decide why compilation was failing.  (The error message occurred later.)